app/src/main/java/com/anysoftkeyboard/gesturetyping/GestureTypingDetector.java

```diff
@@ -117,37 +127,33 @@ public void destroy() {
                 .observeOn(RxSchedulers.mainThread());
     }
 
-    private static int[] generatePath(char[] word, Iterable<Keyboard.Key> keysList, int curvatureSize, WorkspaceData workspaceData) {
+    private static int[] generatePath(char[] word, SparseArray<Keyboard.Key> keysByCharacter, WorkspaceData workspaceData) {
         workspaceData.reset();
         //word = Normalizer.normalize(word, Normalizer.Form.NFD);
         char lastLetter = '\0';
 
         // Add points for each key
         for (char c : word) {
-            c = Dictionary.toLowerCase(c);
+            c = Character.toLowerCase(c);
```

Comment:
So, you're saying it's better to use `Character.toLowerCase`? Should we replace all `Dictionary.toLowerCase` calls to that?

Reply:
It depends, and I think we often would use both. See my other comment on the characters.